source/monitoring.txt

```diff
@@ -0,0 +1,401 @@
+   full list of redacted command events, see the
+   :spec:`MongoDB command logging and monitoring specification </command-logging-and-monitoring/command-logging-and-monitoring.md>`.
+
+Example
+^^^^^^^
+
+This example shows how to make a counter for database commands. The counter
+keeps track of the number of times the driver successfully executes each database
+command and prints this information every time a database command finishes.
+
+To make a counter, follow these steps:
+
+#. Make a class with counter functionality that implements the ``CommandListener`` interface.
+#. Add an instance of the new class to a ``MongoClientSettings`` object.
+#. Configure you ``MongoClient`` instance with the ``MongoClientSettings`` object.
+
+The following code implements these steps:
+
+.. _listener-mongo-client-settings-example:
+
+.. io-code-block::
+
+   .. input:: /includes/monitoring_counter.kt
+      :language: kotlin
+
+   .. output::
+      :language: console
+      :visible: false
+
+      {find=1}
+      {find=2}
+      {find=2, endSessions=1}
+
+For more information on the classes and methods mentioned in this section, see
+the following API documentation:
+
+- `CommandListener <{+core-api+}/com/mongodb/event/CommandListener.html>`__
+- `MongoClientSettings <{+core-api+}/com/mongodb/MongoClientSettings.html>`__
+- `MongoClient <{+api+}/com.mongodb.kotlin.client/-mongo-client/index.html>`__
+- `CommandStartedEvent <{+core-api+}/com/mongodb/event/CommandStartedEvent.html>`__
+- `CommandSucceededEvent <{+core-api+}/com/mongodb/event/CommandSucceededEvent.html>`__
+- `CommandFailedEvent <{+core-api+}/com/mongodb/event/CommandFailedEvent.html>`__
+
+Server Discovery and Monitoring Events
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A server discovery and monitoring (SDAM) event is an event related to a change
+in the state of the MongoDB instance or cluster you have connected the driver to.
+
+The driver defines nine SDAM events and provides the following listener
+interfaces, which listen for three SDAM events each: 
+
+- ``ClusterListener``: topology-related events
+- ``ServerListener``: events related to ``mongod`` or ``mongos`` processes
+- ``ServerMonitorListener``: heartbeat related events
+
+To monitor a type of SDAM event, create a class that implements one of the three
+preceding interfaces and register an instance of that class with your
+``MongoClient`` instance.
+
+For a detailed description of each SDAM event, see the :spec:`MongoDB SDAM monitoring events specification </server-discovery-and-monitoring/server-discovery-and-monitoring-logging-and-monitoring.md>`.
+
+Example
+^^^^^^^
+
+This example shows how to make a listener class that prints a message that lets
+you know if the driver can write to your MongoDB instance.
+
+To make an event that reports write status: 
+
+#. Make a class that tracks and checks cluster description changes, and
+   implements the ``CommandListener`` interface. 
```

Comment:
S: remove comma before dependent clause; also I think "tracks" and "checks" mean pretty much the same thing
```suggestion
#. Make a class that tracks cluster description changes and
   implements the ``CommandListener`` interface. 
```